internal/provisioner/cluster.go

```diff
@@ -88,6 +88,76 @@ func CreateCluster(provider, s3StateStore, size string, zones []string, logger l
 	return nil
 }
 
+// UpgradeCluster upgrades a cluster to the latest recommended production ready k8s version.
+func UpgradeCluster(clusterId, s3StateStore string, logger log.FieldLogger) error {
+	logger = logger.WithField("cluster", clusterId)
+
+	dns := clusterDNS(clusterId)
+
+	// Temporarily look for the kops output directory as a local folder named after
+	// the cluster ID. See above.
+	outputDir := path.Join(clusterRootDir, clusterId)
+
+	// Validate the provided cluster ID before we alter state in any way.
+	_, err := os.Stat(outputDir)
+	if err != nil {
+		return errors.Wrapf(err, "failed to find cluster directory %q", outputDir)
+	}
+
+	terraform := terraform.New(outputDir, logger)
```

Comment:
nit: Not sure if we've already been doing this all over the place but I'm not really a fan of creating variables using the name of an imported package. To me it's very confusing to read e.g. is `terraform.Init()` calling a function in the package `terraform` or a method on the `terraform` variable

Reply:
I started that convention in this repository, but agree with @jwilander that it can be confusing. Perhaps `terraformClient`?

Reply:
Yep, agreed!